server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java

```diff
@@ -247,61 +243,58 @@ public static void updateTabletDataFile(ServerContext context, KeyExtent extent,
       String address, ZooLock zooLock, Set<String> unusedWalLogs, TServerInstance lastLocation,
       long flushId) {
     if (extent.isRootTablet()) {
-      if (unusedWalLogs != null) {
-        updateRootTabletDataFile(context, unusedWalLogs);
-      }
-      return;
+      updateRootTabletDataFile(context, unusedWalLogs);
+    } else {
+      updateForTabletDataFile(context, extent, path, mergeFile, dfv, time, filesInUseByScans,
+          address, zooLock, unusedWalLogs, lastLocation, flushId);
     }
-    Mutation m = getUpdateForTabletDataFile(extent, path, mergeFile, dfv, time, filesInUseByScans,
-        address, zooLock, unusedWalLogs, lastLocation, flushId);
-    MetadataTableUtil.update(context, zooLock, m, extent);
+
   }
 
   /**
    * Update the data file for the root tablet
    */
   private static void updateRootTabletDataFile(ServerContext context, Set<String> unusedWalLogs) {
-    TabletMutator tablet = context.getAmple().mutateTablet(RootTable.EXTENT);
-    unusedWalLogs.forEach(tablet::deleteWal);
-    tablet.mutate();
+    if (unusedWalLogs != null) {
+      TabletMutator tablet = context.getAmple().mutateTablet(RootTable.EXTENT);
+      unusedWalLogs.forEach(tablet::deleteWal);
+      tablet.mutate();
+    }
   }
 
   /**
    * Create an update that updates a tablet
    *
-   * @return A Mutation to update a tablet from the given information
    */
-  private static Mutation getUpdateForTabletDataFile(KeyExtent extent, FileRef path,
+  private static void updateForTabletDataFile(ServerContext context, KeyExtent extent, FileRef path,
       FileRef mergeFile, DataFileValue dfv, String time, Set<FileRef> filesInUseByScans,
       String address, ZooLock zooLock, Set<String> unusedWalLogs, TServerInstance lastLocation,
       long flushId) {
-    Mutation m = new Mutation(extent.getMetadataEntry());
+
+    TabletMutator tablet = context.getAmple().mutateTablet(extent);
 
     if (dfv.getNumEntries() > 0) {
-      m.put(DataFileColumnFamily.NAME, path.meta(), new Value(dfv.encode()));
-      TabletsSection.ServerColumnFamily.TIME_COLUMN.put(m, new Value(time.getBytes(UTF_8)));
-      // stuff in this location
+      tablet.putFile(path, dfv);
+      tablet.putTime(time);
+
       TServerInstance self = getTServerInstance(address, zooLock);
-      self.putLastLocation(m);
-      // erase the old location
-      if (lastLocation != null && !lastLocation.equals(self))
-        lastLocation.clearLastLocation(m);
```

Comment:
Eventually this clearLastLocation method will become dead code and can be removed.

Reply:
should it be deprecated?

Reply:
No, its not public API.  So when nothing is using it, it can be deleted.

Reply:
So how are those types of things cleaned up.  Is there something run periodically that identifies unused code?

Reply:
I think there are tools that will find unused public methods, but I am not sure.  I think in the past someone has run a tool like that and removed the code that made sense.  But I am not sure what the tool was.

Reply:
IntelliJ has a great built in Analysis tool for unused code.  That is what I use.

Reply:
So funny... I lterally just a few moments ago saw one faded out and remembered .."oh yeah, intellj does this automatically"... so next question is if something is noticed here not being used should we put a ticket in to remove it?  Like ```
MetadataTableUtil.getBulkFilesLoaded(ServerContext context, AccumuloClient client,
      KeyExtent extent, long tid)
```
This signature with 'tid' is not used.

Reply:
I usually do a separate refactor PR like:  Remove unused code.  It is easier to review if its all just unused code being removed.